tests/data_group_test.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+import scipp as sc
+
+
+def test_add():
+    x = sc.arange('x', 4, unit='m')
+    dg1 = sc.DataGroup({'a': x})
+    dg2 = sc.DataGroup({'a': x, 'b': x})
```

Comment:
I know we are missing many tests, but we could at least check that we can insert Variables and DataArrays with different dimension lengths, since this is kind of the main feature of a DataGroup?

Reply:
See https://github.com/scipp/scipp/tree/datagroup-tests, which I am working on now.